x-pack/platform/plugins/private/upgrade_assistant/server/lib/data_streams/data_stream_migration_service.ts

```diff
@@ -333,7 +333,36 @@ export const dataStreamMigrationServiceFactory = ({
       }
     },
 
-    async readonlyIndices(indices: string[]) {
+    async readonlyIndices(dataStreamName: string, indices: string[]) {
+      try {
+        const { data_streams: dataStreamsDeprecations } = await esClient.migration.deprecations({
+          filter_path: `data_streams`,
+        });
+
+        const deprecationsDetails = dataStreamsDeprecations[dataStreamName];
+        if (!deprecationsDetails || !deprecationsDetails.length) {
+          throw error.cannotGrabMetadata(`Could not grab metadata for ${dataStreamName}.`);
+        }
+        // Find the first deprecation that has reindex_required set to true
+        const deprecationDetails = deprecationsDetails.find(
+          (deprecation) => deprecation._meta!.reindex_required
+        );
+        if (!deprecationDetails) {
+          throw error.cannotGrabMetadata(`Could not grab metadata for ${dataStreamName}.`);
+        }
+
+        const rollOverResponse = await esClient.transport.request({
```

Comment:
nit: we can make this more optimal by checking whether any index in `indices` is the last index listed from get data stream API ([docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-data-stream.html#get-data-stream-api-response-body)) and only calling rollover if that is true.